src/thunderbots/software/ai/hl/stp/tactic/shadow_enemy_tactic.cpp

```diff
@@ -0,0 +1,97 @@
+}
+
+void ShadowEnemyTactic::updateParams(const Evaluation::EnemyThreat &enemy_threat,
+                                     const Field &field, const Team &friendly_team,
+                                     const Team &enemy_team, double shadow_distance,
+                                     bool enemy_team_can_pass)
+{
+    this->enemy_threat        = enemy_threat;
+    this->field               = field;
+    this->friendly_team       = friendly_team;
+    this->enemy_team          = enemy_team;
+    this->shadow_distance     = shadow_distance;
+    this->enemy_team_can_pass = enemy_team_can_pass;
+}
+
+double ShadowEnemyTactic::calculateRobotCost(const Robot &robot, const World &world)
+{
+    if (!enemy_threat)
+    {
+        return 0;
+    }
+    // Prefer robots closer to the enemy being shadowed
+    // We normalize with the total field length so that robots that are within the field
+    // have a cost less than 1
+    double cost = (robot.position() - enemy_threat->robot.position()).len() /
+                  world.field().totalLength();
+    return std::clamp<double>(cost, 0, 1);
+}
+
+void ShadowEnemyTactic::calculateNextIntent(IntentCoroutine::push_type &yield)
+{
+    MoveAction move_action = MoveAction();
+    StopAction stop_action =
+        StopAction(StopAction::ROBOT_STOPPED_SPEED_THRESHOLD_DEFAULT, true);
+    do
+    {
+        if (!enemy_threat)
+        {
+            yield(stop_action.updateStateAndGetNextIntent(*robot, false));
+        }
+
+        Robot enemy_robot = enemy_threat->robot;
+        // If we think the enemy team can pass, and if we have identified a robot that can
+        // pass to the robot we are shadowing, we block the pass rather than block the
+        // net. Otherwise, we just block the net
+        if (enemy_team_can_pass && enemy_threat->passer)
+        {
+            Vector enemy_to_passer_vector =
+                enemy_threat->passer->position() - enemy_robot.position();
+            Point position_to_block_pass =
+                enemy_robot.position() +
+                enemy_to_passer_vector.norm(this->shadow_distance);
+            yield(move_action.updateStateAndGetNextIntent(
+                *robot, position_to_block_pass, enemy_to_passer_vector.orientation(), 0));
+        }
+        else
+        {
+            auto best_enemy_shot_opt = Evaluation::calcBestShotOnFriendlyGoal(
+                field, friendly_team, enemy_team, enemy_robot, ROBOT_MAX_RADIUS_METERS,
+                {*robot});
+            Vector enemy_shot_vector = Vector(0, 0);
+            if (best_enemy_shot_opt)
+            {
+                enemy_shot_vector = best_enemy_shot_opt->first - enemy_robot.position();
+            }
+            else
+            {
+                enemy_shot_vector = field.friendlyGoal() - enemy_robot.position();
+            }
+            Point position_to_block_shot =
+                enemy_robot.position() + enemy_shot_vector.norm(this->shadow_distance);
```

Comment:
Just a thought, maybe we should be shadowing with our robot in the 'shot cone' of the enemy robot

Reply:
That is an option, but our current "shot cone" calculations put the robots as far away from the enemy while still blocking the cone. By just setting a small shadow distance we should be fine. What do you think?

Reply:
I'm concerned of robots being in useless positions because of the constant shadow distance. 

Near the net/out-of-bounds I think we would see issues with this.

Reply:
But isn't being closer to the enemy always better? What's an example of a useless position we might end up in? (Might merge before we resolve this but lets keep this discussion going so we can come back and change things after. Just don't want this PR to sit for too long).

Reply:
Started a thread on slack

Reply:
Added comments to slack.